spec/lib/whitehall_importer/import_spec.rb

```diff
@@ -163,8 +165,20 @@
                       edition: build(:edition),
                     ))
 
-      expect { described_class.call(build(:whitehall_migration_document_import)) }
+      document_import = create(:whitehall_migration_document_import, document: nil)
+      expect { described_class.call(document_import) }
         .to raise_error(WhitehallImporter::IntegrityCheckError)
     end
+
+    it "does not update the document import if the transaction fails" do
+      document_import = create(:whitehall_migration_document_import)
+      updated_at = document_import.updated_at
+      document_import.updated_at = 2.days.from_now
+      allow(document_import).to receive(:update!).and_raise("forced error")
+
+      expect { described_class.call(document_import) }.to raise_error("forced error")
+      expect(document_import.changed?).to be false
+      expect(document_import.updated_at.to_i).to eq(updated_at.to_i)
```

Comment:
I was a bit puzzled by the to_i, I tried it on my machine and this passes with `expect(document_import.updated_at).to eq(updated_at)` so I don't think that's needed.

Reply:
I added `to_i` as it wasn't passing for me locally.  I got:
```
expected: 2020-01-23 15:26:25.365678900 +0000
            got: 2020-01-23 15:26:25.365678000 +0000
```

I found a few solutions that suggested using `to_i` to overcome it e.g. https://stackoverflow.com/questions/20403063/trouble-comparing-time-with-rspec

Reply:
Ah I see - sorry I didn't clock that as a precision check. Maybe I got lucky when I ran the test 🤔 time precision is a pain.

We've got around this in other parts of the suite by using times that are fixed to a second. I'd suggest switching `2.days.from_now` to `Date.tomorrow.noon` - which is consistent with what we do in the factories: https://github.com/alphagov/content-publisher/commit/6edd08fcacd10a16d18124840799f276da7e493e so then you'll not need to convert to integer.

Reply:
Seems like just passing a specific date to the factory made it pass without the need for `to_i` 👍

Reply:
Seems like just passing a specific date to the factory made it pass without the need for to_i 👍